src/AppShed/Extensions/SpreadsheetBundle/Controller/ReadController.php

```diff
@@ -146,23 +143,42 @@ public function documentAction(Request $request)
                             $link->setScreenLink($innerScreen);
                         } else {
                             if (!empty($value)) {
+                                $mapCreated = 0;
                                 if ($name == $address ) {
 
-                                    $geo = $geoService->getGeo($value);
+                                    $geoResponse = $this->geoClient->get(
+                                        false,
+                                        [
+                                            'query' => [
+                                                'sensor'=>false,
+                                                'address' => $value
+                                            ]
+                                        ]
+                                    );
+
+                                    if($geoResponse->getStatusCode()==200){
```

Comment:
It was better with the service!

You should make the Client the argument to the service, inside the service you should have the `try{}catch` and the checks that everything is correct, so that the `Controller` is just really simple